templates/template_processor.go

```diff
@@ -160,15 +160,21 @@ func createOutputDir(dir string, options *config.BoilerplateOptions) error {
 	return os.MkdirAll(destination, 0777)
 }
 
-// Compute the path where the given file, which is in templateFolder, should be copied in outputFolder
-func outPath(file string, templateFolder string, outputFolder string) (string, error) {
-	// TODO process template syntax in paths
-	templateFolderAbsPath, err := filepath.Abs(templateFolder)
+// Compute the path where the given file, which is in templateFolder, should be copied in outputFolder. If the file
+// path contains boilerplate syntax, use the given options and variables to render it to determine the final output
+// path.
+func outPath(file string, options *config.BoilerplateOptions, variables map[string]string) (string, error) {
+	templateFolderAbsPath, err := filepath.Abs(options.TemplateFolder)
 	if err != nil {
 		return "", errors.WithStackTrace(err)
 	}
 
-	fileAbsPath, err := filepath.Abs(file)
+	interpolatedFilePath, err := renderTemplate(file, file, variables, options.OnMissingKey)
```

Comment:
Notably, you're using the `renderTemplate()` function as is. Cool! So, since we're now rendering paths on the file system, I suppose we're still expecting UTF-8 on a *nix box and likely Windows boxes as well? Also, do we need to filter out a `/` somehow since filenames shouldn't be allowed to have `/`?